sdk/communication/communication-short-codes/src/shortCodesClient.ts

```diff
@@ -65,7 +55,7 @@ export class ShortCodesClient {
     credentialOrOptions?: KeyCredential | TokenCredential | ShortCodesClientOptions,
     maybeOptions: ShortCodesClientOptions = {}
   ) {
-    const { url, credential } = parseClientArguments(connectionStringOrUrl, credentialOrOptions);
+    const { url } = parseClientArguments(connectionStringOrUrl, credentialOrOptions);
```

Comment:
@joheredi  Hello! I've been getting 401s since trying to update the http client. Anything you can catch? My guess is the credentials are missing?

Reply:
Yeah, it looks like you are no longer creating and adding the CommunicationAuthPolicy. Why was `const authPolicy = createCommunicationAuthPolicy(credential);` removed?

Reply:
I think you'll need to re-write your communicationAuthPolicy.ts to align with the new PipelinePolicies. Something like this should do

```typescript

import { PipelinePolicy, PipelineRequest } from "@azure/core-rest-pipeline";
export const communicationAccessKeyCredentialPolicyName = "CommunicationAccessKeyCredentialPolicy";
/**
 * Creates an HTTP pipeline policy to authenticate a request using a `KeyCredential`.
 * @hidden
 *
 * @param credential - The key credential.
 */
export const createCommunicationAccessKeyCredentialPolicy = (
  credential: KeyCredential
): PipelinePolicy => {
  return {
    name: communicationAccessKeyCredentialPolicyName,
    sendRequest: async (req, next) => {
      const signedRequest = await signRequest(credential, req);
      return next(signedRequest);
    }
  };
};

async function signRequest(
  accessKey: KeyCredential,
  request: PipelineRequest
): Promise<PipelineRequest> {
  const verb = request.method.toUpperCase();
  const utcNow = new Date().toUTCString();
  const contentHash = await shaHash(getBody(request) || "");
  const dateHeader = "x-ms-date";
  const signedHeaders = `${dateHeader};host;x-ms-content-sha256`;

  const url = URLBuilder.parse(request.url);
  const query = url.getQuery();
  const urlPathAndQuery = query ? `${url.getPath()}?${query}` : url.getPath();
  const port = url.getPort();
  const hostAndPort = port ? `${url.getHost()}:${port}` : url.getHost();

  const stringToSign = `${verb}\n${urlPathAndQuery}\n${utcNow};${hostAndPort};${contentHash}`;
  const signature = await shaHMAC(accessKey.key, stringToSign);

  if (isNode) {
    request.headers.set("Host", hostAndPort || "");
  }

  request.headers.set(dateHeader, utcNow);
  request.headers.set("x-ms-content-sha256", contentHash);
  request.headers.set(
    "Authorization",
    `HMAC-SHA256 SignedHeaders=${signedHeaders}&Signature=${signature}`
  );

  return request;
}

function getBody({ body }: PipelineRequest): string {
  return body === undefined || typeof body !== "string" ? "" : body;
}

```


And then your createCommunicationAuthPolicy

```typescript
import { bearerTokenAuthenticationPolicy, PipelinePolicy } from "@azure/core-rest-pipeline";

export const createCommunicationAuthPolicy = (
  credential: KeyCredential | TokenCredential
): PipelinePolicy => {
  if (isTokenCredential(credential)) {
    return bearerTokenAuthenticationPolicy({
      scopes: "https://communication.azure.com//.default",
      credential
    });
  } else {
    return createCommunicationAccessKeyCredentialPolicy(credential);
  }
};
```

and lastly add the policy to the pipeline here.

```typescript
this.client.addPolicy(credential);
```

Reply:
I see that the policies are in communication-common. So you may need to add a new policy using the code I shared instead of just updating the existing files. Once all the communication packages are migrated the old ones can be removed.

Let me know if you have any questions, I'll be happy to help